lib/lintSource.js

```diff
@@ -66,6 +66,20 @@ module.exports = async function lintSource(stylelint, options = {}) {
 	const config = configForFile.config;
 	const existingPostcssResult = options.existingPostcssResult;
 
+	if (options.cache) {
+		if (!stylelint._fileCache._hashOfConfig) {
+			stylelint._fileCache.calcHashOfConfig(config);
+		}
+
+		if (options.filePath && !stylelint._fileCache.hasFileChanged(options.filePath)) {
+			return options.existingPostcssResult
+				? Object.assign(options.existingPostcssResult, {
```

Comment:
[suggest] The code can be shorter thanks to a local variable:

```suggestion
			return existingPostcssResult
				? Object.assign(existingPostcssResult, {
```

Reply:
Thanks. Changed use local variable. [1bf3607](https://github.com/stylelint/stylelint/pull/6356/commits/1bf36073eb112b6819a42828695a1ed58576d1f2)